dokka-integration-tests/gradle/src/main/kotlin/org/jetbrains/dokka/it/gradle/junit/TestedVersions.kt

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package org.jetbrains.dokka.it.gradle.junit
+
+import org.jetbrains.dokka.it.gradle.utils.SemVer
+import org.jetbrains.dokka.it.optionalSystemProperty
+import org.jetbrains.dokka.it.systemProperty
+
+
+/**
+ * The parameters used to run each invocation of a [DokkaGradlePluginTest] test.
+ *
+ * [DokkaGradlePluginTestExtension] generates all possible values, and creates a test invocation
+ * for each valid combination.
+ *
+ * To access the specific values used for a test invocation, add a [TestedVersions]
+ * parameter in the test function.
+ */
+data class TestedVersions(
+    /** Kotlin Gradle Plugin version. */
+    val kgp: SemVer,
+    /** Gradle version. */
+    val gradle: SemVer,
```

Comment:
nitpick: for some reason, I feel like `gradle` version should go first :)

Reply:
I will order them alphabetically